Detectors/TRD/base/include/TRDBase/Calibrations.h

```diff
@@ -66,37 +65,57 @@ class Calibrations
   //
   int const getTimeStamp() { return mTimeStamp; }
   void setTimeStamp(long timestamp) { mTimeStamp = timestamp; }
-  void setCCDB(int calibrationobjecttype);
-  void setCCDBForSimulation() { setCCDB(kSimulation); };
-  void setCCDBForReconstruction() { setCCDB(kReconstruction); };
-  void setCCDBForCalibration() { setCCDB(kCalibration); };
-  //// cant send auto to setCCDB, as a temporary measure, until I can figure something else out. There are 3 formats of getting CCDB parameters defined by the enum at the top.
+  void setCCDB(int calibrationobjecttype, long timestamp);
+  void setCCDBForSimulation(long timestamp) { setCCDB(kSimulation, timestamp); };
+  void setCCDBForReconstruction(long timestamp) { setCCDB(kReconstruction, timestamp); };
+  void setCCDBForCalibration(long timestamp) { setCCDB(kCalibration, timestamp); };
   //
-  //  float getGainMeanRMS();
-  //  float getT0MeanRMS();
-  double getVDrift(long timestamp, int roc, int col, int row) const;
-  double getT0(long timestamp, int roc, int col, int row) const;
-  double getExB(long timestamp, int roc, int col, int row) const;
-  double getGainFactor(long timestamp, int roc, int col, int row) const;
+  double getVDrift(int roc, int col, int row) const;
+  double getT0(int roc, int col, int row) const;
+  double getExB(int roc) const;
+  double getGainFactor(int roc, int col, int row) const;
+  double getPadGainFactor(int roc, int col, int row) const;
```

Comment:
I think getGainFactor should return the combined gain factor, i.e. the combination of the pad-by-pad, once per run and the chamber-wise, run-wise gain factors.

Reply:
@tdietel yes it does, but @jolopezl specifically wanted the pure pad value, hence that method.
So as I understand it getPadGainFactor should be removed as it will give the wrong value?

Reply:
@tdietel, @bazinski, I didn't ask for a explicit value. I asked you to confirm if these two method were equivalent to what was used in AliRoot:

float padgain = calGainFactorDet->GetValue(det) * calGainFactorROC->GetValue(col, row)

So that, when I use it, I get the same value and we get consistent results.

Reply:
@jolopezl Sorry my bad wording. Yes you must use the combined value so getGainFactor(int int int) combining the infrequent calibration giving the pad wise value and the frequent calibration giving the cahmber wise value.